### PR TITLE
docs: remove unused go.mod

### PR DESCRIPTION
**Pull request by thaJeztah**
- follow-up to https://github.com/distribution/distribution/pull/4101

### CODE REVIEW

File: docs/go.mod [hunk 1]
```diff
@@ -1,9 +1,3 @@
 module github.com/distribution/distribution/docs
 
-go 1.21.1
-
-require (
-	github.com/google/docsy v0.7.1 // indirect
-	github.com/imfing/hextra v0.5.0 // indirect
-	github.com/thegeeklab/hugo-geekdoc v0.41.2 // indirect
-)
+go 1.20
```
Comment (thaJeztah) [inline docs/go.mod:3]: We could even remove the version, but looks like go mod tidy wants to add "something" back. Technically "any" version would work, so we could set it to something really low.

**Review by milosgajdos — APPROVED** [review]
Makes sense.

**Comment by Jamstah** [comment]
This is probably just my lack of go experience, but, why do we have a go.mod in the docs package at all when there is no code there?

**Comment by thaJeztah** [comment]
Perhaps it needs a comment; this is used to facilitate Hugo's feature to vendor documentation from multiple repositories; Hugo has a feature to get documentation "modules" from repositories, and it uses the Go modules logic for that; https://gohugo.io/hugo-modules/

The docker documentation currently uses this to combine docs from multiple sources (see https://github.com/docker/docs/blob/main/go.mod)

I guess the added benefit is that vendoring `distribution/distribution` will skip the `docs` directory (as it's not in the main module)

@dvdksn may be able to provide more details if needed

**Review by Jamstah — APPROVED** [review]
No harm and supports other workflows, so why not.

A comment somewhere would be nice, maybe in the README.md for the docs?

**Comment by dvdksn** [comment]
Just checked and the `go.mod` and `go.sum` are not needed in the docs directory on this repo. They were generated when I was experimenting with different themes for the Hugo site here, but the theme we're using now doesnt use hugo modules. So both files can be safely deleted.

**Comment by milosgajdos** [comment]
@thaJeztah do you wanna delete those files as per @dvdksn comment?

**Comment by milosgajdos** [comment]
Ping @thaJeztah RE: my and @dvdksn's last comments

**Comment by thaJeztah** [comment]
whoops, missed your ping; updated and removed the `go.mod` and `go.sum` from the docs directory 👍

**Pull request merged**